otherlibs/unix/unix_unix.ml

```diff
@@ -962,13 +965,13 @@ end = struct
     | Some mutex -> Mutex.protect mutex f
 end
 
-let popen_mutex = lazy (Mutex.create ())
+let popen_mutex = Mutex.create ()
 
 let open_proc prog args envopt proc input output error =
   let pid =
     create_process_gen prog args envopt input output error in
-  Mutex.protect (Lazy.force popen_mutex) (fun () ->
-    Hashtbl.add popen_processes proc pid)
+  Mutex.protect popen_mutex (fun () ->
+    Hashtbl.add (magic_uncontended popen_processes) proc pid)
```

Comment:
The `popen_mutex` is used to protect accesses to `popen_processes`.  Instead of using magic here, another alternative would perhaps be to use the capsule API.  I guess that might complicate things here as the capsule API is currently in `stdlib_alpha` &mdash; perhaps we should revisit this once the capsule API is in `stdlib`.

Reply:
See #3698